### DISCUSSION TITLE
Evaluation of gsm on study data as of v0.1.0 (Mar11)

### DISCUSSION THREAD

**Discussion by gwu05** [Gilead-Public/gsm#280] [Show and tell]
Hi, I evaluated our code on an active study to see if any fell out based on the current version for release. Overall, worked well and I was able to get the results I wanted. Thinking we can discuss and parse which should be issues.

Here's what I found:

- vThreshold for dfAE : for simplicity, I think we can change to one parameter

- CreateRDSL in clindata package is going to be an essential function for gsm release as it creates dfRDSL
- Can we consider creating a custom mapping functionality for CreateRDSL?
- Will likely need to re-visit CreateRDSL for
       - not require all datasets to run
       - be able to handle dates flexibly [ RGMNDTC (dmy) vs RGMNDTN (ymd) from IXRSRAND]
       - possible re-visit TimeOnTreatment rules ( dfEx by default didn't work well to make RDSL vs ADSL as got time on-treatment pretty wrong)
       - possibly consider multiple treatment scenarios (SDRGCOMP1, SDRGCOMP2, etc.)

- Looks like IE will require stacking the IE datasets, and expected values default should probably be "N","Y" rather than 0/1
- The output of IE$dfFlagged should be ordered by flagged first
- We can calculate p-values for this IE assessment, but need to create the analyze function; essentially take the proportion overall and do a binom.test. The way we flag the assessment basically shouldn't preclude the calculation of the p-value (i.e. we can still flag if any IE missing is detected but still provide p-values for sorting)

- Consents may have multiple amendments for main consent, and that's where the before rand date may not work as well (may need to be re-visited)
- The outcome of Consents derivation looks to be flipped; we should be looking for missing consent to count as 1
- Like above we can likely add p-value for Consents assessment

**Comment by jwildfire** [comment]
> * vThreshold for dfAE : for simplicity, I think we can change to one parameter

vThreshold becomes nThreshold and value for poission gets passed to flag as c(-1*nThrehsold, nThrehsold). @dsanders2gilead to create issue. 

> * CreateRDSL in clindata package is going to be an essential function for gsm release as it creates dfRDSL
> * Can we consider creating a custom mapping functionality for CreateRDSL?

@gwu05 to file issue in clindata. We can consider migrating to gsm. 

> * Will likely need to re-visit CreateRDSL for
>   - not require all datasets to run
>   - be able to handle dates flexibly [ RGMNDTC (dmy) vs RGMNDTN (ymd) from IXRSRAND]
>   - possible re-visit TimeOnTreatment rules ( dfEx by default didn't work well to make RDSL vs ADSL as got time on-treatment pretty wrong)
>   - possibly consider multiple treatment scenarios (SDRGCOMP1, SDRGCOMP2, etc.)

@jwildfire Create a wiki page in clindata tracking some study level customizations from pilots. 

> * Looks like IE will require stacking the IE datasets, and expected values default should probably be "N","Y" rather than 0/1

Already have an issue for this. 

> * The output of IE$dfFlagged should be ordered by flagged first

@gwu05 to file issue for v0.3

> * We can calculate p-values for this IE assessment, but need to create the analyze function; essentially take the proportion overall and do a binom.test. The way we flag the assessment basically shouldn't preclude the calculation of the p-value (i.e. we can still flag if any IE missing is detected but still provide p-values for sorting)

@gwu05 to file issue (no milestone)

> * Consents may have multiple amendments for main consent, and that's where the before rand date may not work as well (may need to be re-visited)
> * The outcome of Consents derivation looks to be flipped; we should be looking for missing consent to count as 1

@gwu05 to file issue for v0.3. Assign to @dsanders2gilead 

> * Like above we can likely add p-value for Consents assessment

@gwu05 to make combined issue for both assessments
